src/shared/cpu-set-util.c

```diff
@@ -133,6 +133,28 @@ int cpu_set_add_all(CPUSet *a, const CPUSet *b) {
         return 1;
 }
 
+int cpu_set_copy(CPUSet *dst, CPUSet *src) {
+        cpu_set_t *t;
+
+        assert(src);
+        assert(dst);
+
+        if(src->set == NULL || src->allocated == 0)
+                return -EFAULT;
```

Comment:
Hmm, it should be OK to copy an empty CPUSet. SO I think this should just call `cpu_set_reset(dst)` instead.

Please don't use `== NULL`. For pointers and booleans, our coding style wants `!`.